### ISSUE TITLE
How to setup Rhubarb server through express

### ISSUE THREAD

**Issue by dreammonkey** [oguzeroglu/Rhubarb#2]
Is there a way to setup Rhubarb together with express like it is done with socet.io ?

for example:

```javascript
var app = require('express')();
var server = require('http').Server(app);
var rhubarb = require('rhubarb-js')(server);
server.listen(80);
```

**Comment by oguzeroglu** [comment]
No, sadly not. However what exactly are you trying to achieve? Maybe I could help you :)

**Comment by oguzeroglu** [comment]
AFAIK Express is used for web servers/REST APIs. Rhubarb is designed for WebSocket based game servers. You don't usually want to host HTML files or handle REST within your game server. Game servers would only take care of game related logic (physics calculations, position handling etc.).

Socket.io is rather a general purpose WebSocket library. So if you're using Rhubarb maybe you should not be doing this anyway.

**Comment by dreammonkey** [comment]
So you are proposing to setup 2 servers ? One (express) for hosting assets and authenticating users and one (websocket) for all the (multiplayer-)game logic ?

I've done a similar setup in the past using the Colyseus framework based on this example:
https://github.com/endel/tanx/blob/master/app.js

We are thinking about using Rhubarb for our serverside game logic, but our biggest struggle currently is figuring out a way how to render our game scene (three.js) on the server. Do I understand correctly that the ROYGBIV engine is actually made with this in mind ? I had a quick look, but I don't fully understand how to set it up, maybe a getting started section might be handy... ?

**Comment by oguzeroglu** [comment]
Yes this is exactly what I'm proposing. Since you want your game server as fast as possible, you don't want to mix it with HTML hosting, authentication logic. You should run your game server separately from all that.

You don't actually want to render game scene on the server side. What you want to do is that, lets say if you have a physics engine, you never want the graphical representations of those in your server side. You only care about the numbers -> Position and Quaternion of each player. That's why you should not integrate ThreeJS into your game servers. 

Since each position/quaternion data is send 60 times per second from the client, Rhubarb helps optimizing the client performance and bandwidth usage (using webworkers, binary data instead of JSON, transferables etc.) and also helps sharing your game related protocol definitions between the server and the client. It's up to you to decide how you want to integrate your game logic into it. But ideally THREE.JS should never be a part of the server.

ROYGBIV uses CannonJS as physics engine, so if I implemented a multiplayer shooter, I'd use a NodeJS implementation of CannonJS in the server to take care of player positions. However it's on the roadmap of ROYGBIV to expose some server-side utilities as well.

TLDR -> Rhubarb is just a networking library that aims giving client side 3D rendering more space by being smart (that helps you having 60 FPS). It's up to you to decide what numbers/texts you want to transfer with it :) 

>  maybe a getting started section might be handy... ?

ROYGBIV is still under development, I'll document it as soon as I'm done. Rhubarb has a Getting Started page and a tutorial though: https://github.com/oguzeroglu/Rhubarb/wiki/Getting-started

I already implemented a multiplayer card game (a Cards Against Humanity clone) with Rhubarb and ROYGBIV: https://aykir.io
In this game I exchange client information and selected cards using Rhubarb.
